Deploying telemetry module with the lifecycle manager (#110)
kyma-project/telemetry-manager: docs/deploying_module_template.md

> ## Deploying `ModuleTemplate` with the Lifecycle Manager

Reviewer: I'm wondering if we should have this in the README.md since it is kind of related to the development flows that are described there. At least we should link this document there. If you keep this file as it is, start with a H1 (#).

> 1. Bundle the `telemetry` module as an OCI image and pusht it to a registery. You can push the OCI image either to a local k3d registry or to a remote registry.

Reviewer 2: ```suggestion
1. Bundle the `telemetry` module as an OCI image and pusht it to a registry. You can push the OCI image either to a local k3d registry or to a remote registry.
```

I really like that you're listing the two options with a bullet list. Way to go! 👍

    
    *   Using a local k3d registry:
        
        > **NOTE:** The following sub-steps assume that you have created a k3d cluster using the Kyma CLI with `kyma provision k3d`. This command creates a k3d registry which is exposed on port `5001`.
        

>         *   Create the telemetry module, bundle it as an OCI image and push it to the local k3d registry:

Reviewer 2: ```suggestion
        1. Create the Telemetry module, bundle it as an OCI image and push it to the local k3d registry:
```

I assume that these steps must be done in the specified order, so I'd use a numbered list.

            ```shell
            kyma alpha create module --name kyma-project.io/module/telemetry --version 0.0.1 --channel alpha --default-cr ./config/samples/operator_v1alpha1_telemetry.yaml --registry localhost:5001 --insecure
            ```
            The auto-generated `template.yaml` file contains the `ModuleTemplate`.

>         *   In the `template.yaml` file, change the `spec.descriptor.component.repositoryContexts.baseUrl` field from `http://localhost:5001` to `http://k3d-kyma-registry.localhost:5000`. This allows the [lifecycle-manager](https://github.com/kyma-project/lifecycle-manager/tree/main) to pull the module image from inside the k3d cluster.

Reviewer 2: ```suggestion
        2. In the `template.yaml` file, change the `spec.descriptor.component.repositoryContexts.baseUrl` field from `http://localhost:5001` to `http://k3d-kyma-registry.localhost:5000`. Now the [lifecycle-manager](https://github.com/kyma-project/lifecycle-manager/tree/main) can pull the module image from inside the k3d cluster.
```

I hate "allow" for anything that wasn't forbidden before. Usually, we can replace it with "enable", and then again, "enable" often can be simplified with "xy can..." :D

    *   Using a remote registry:
        

>         > **NOTE** In this example, we use the Google Container Registry (GCR) as our remote registry. The following sub-steps assume that you are a member of the Google Cloud Platform (GCP) project called `sap-kyma-huskies-dev`.

Reviewer 2: ```suggestion
        > **NOTE** The following example uses the Google Container Registry (GCR) as remote registry. The following sub-steps assume that you are a member of the Google Cloud Platform (GCP) project called `sap-kyma-huskies-dev`.
```

>         > **NOTE** In this example, we use the Google Container Registry (GCR) as our remote registry. The following sub-steps assume that you are a member of the Google Cloud Platform (GCP) project called `sap-kyma-huskies-dev`.

Author: After discussion, we decided to place here a generic command for bundling the telemetry module as an OCI image and pushing it to a remote registry. The details for pushing the image to GCR in the `sap-kyma-huskies-dev` will be moved to the interal knowledge hub for the Huskies team.

>         *   In order to be able to create OAuth2 access tokens, ensure [here](https://console.cloud.google.com/iam-admin/iam?project=sap-kyma-huskies-dev) that you are assigned the `Service Account Token Creator` role.

Reviewer 2: ```suggestion
        1. To create OAuth2 access tokens, ensure [here](https://console.cloud.google.com/iam-admin/iam?project=sap-kyma-huskies-dev) that you are assigned the `Service Account Token Creator` role.
```

I assume that these steps must be done in the specified order, so I'd use a numbered list.

>         *   Authorize `gcloud` to access the Cloud Platform with Google user credentials:

Reviewer 2: ```suggestion
        2. Authorize `gcloud` to access the Cloud Platform with Google user credentials:
```

            ```shell
            gcloud auth login
            ```
        

>         *   Set the `project` property to `sap-kyma-huskies-dev`:

Reviewer 2: ```suggestion
        3. Set the `project` property to `sap-kyma-huskies-dev`:
```

            ```shell
            gcloud config set project sap-kyma-huskies-dev
            ```
        

>         * Set an environment variable (`GCR_TOKEN`) with the access token for the Service Account calld `telemetry-module-sa`:

Reviewer 2: ```suggestion
        4. Set an environment variable (`GCR_TOKEN`) with the access token for the Service Account called `telemetry-module-sa`:
```
